modules/_labs/header/react/lib/GlobalHeader.tsx

```diff
@@ -16,7 +16,7 @@ export interface GlobalHeaderProps {
   /**
    * The function called when the responsive menu icon is clicked.
    */
-  onMenuClick?: (e: React.SyntheticEvent) => void;
+  onMenuClick?: (e: React.MouseEvent<HTMLButtonElement>) => void;
```

Comment:
My general rule for typing these was to provide a more specific typing if I was fairly certain about the element being passed in the click event, and more general if the element was ambiguous. We can definitely de-specify some of these types if wanted though.

Reply:
That is a good thing to point out. Being more specific does allow an app developer to get a more accurate `target` interface. It would mean that changing the element that we use would be a breaking change.

Is the interface of the target element important here?

```ts
let button = document.createElement('button')
let div = document.createElement('div')

div = button // TS error: HTMLButtonElement is missing 'align'
button = div // TS error: missing button props

type OmittedProps = keyof Omit<HTMLButtonElement, keyof HTMLDivElement>
//  "form" | "disabled" | "formAction" | "formMethod" | "formNoValidate" | "formTarget" | "name" | "type" | "value" | "formEnctype" | "labels" | "validationMessage" | "validity" | "willValidate" | "checkValidity" | "reportValidity" | "setCustomValidity"
```

Almost all of those extra properties have to do with being part of a form. I think specifying the element is extremely helpful for form fields. I'm not sure it is useful outside the context of a form field.

For example, the MenuItem - the element type might depend on the context the menu is used in. A navigation menu might have buttons as the target element, but an ARIA role of `menuitem` while a combobox might have `li` with a ARIA role of `option`. We might make a decision on the element based on some inherit properties of the element, but choose a role based on the correct ARIA context.

In some generic contexts, we may want to do either:
* Use `HTMLElement` because we're making no guarantees on the specific element interface
* Use a generic interface defaulting to `HTMLElement`: `<T = HTMLElement>`